app/experimenter/nimbus-ui/src/components/AppLayoutSidebarLocked/index.tsx

```diff
@@ -0,0 +1,202 @@
+  );
+  return newMap;
+};
+
+type AppLayoutSidebarLockedProps = {
+  testid?: string;
+  children: React.ReactNode;
+  status: StatusCheck;
+  analysis?: AnalysisData;
+  analysisLoadingInSidebar?: boolean;
+  analysisError: Error | undefined;
+  primaryProbeSets:
+    | (getExperiment_experimentBySlug_primaryProbeSets | null)[]
+    | null;
+  secondaryProbeSets:
+    | (getExperiment_experimentBySlug_secondaryProbeSets | null)[]
+    | null;
+} & RouteComponentProps;
+
+export const AppLayoutSidebarLocked = ({
+  children,
+  testid = "AppLayoutSidebarLocked",
+  status,
+  analysis,
+  analysisLoadingInSidebar = false,
+  analysisError,
+  primaryProbeSets,
+  secondaryProbeSets,
+}: AppLayoutSidebarLockedProps) => {
+  const { slug } = useParams();
+  const primaryMetrics = probesetToMapping(primaryProbeSets || []);
+  const secondaryMetrics = probesetToMapping(secondaryProbeSets || []);
+  const sidebarKeys = [
+    "monitoring",
+    "overview",
+    "results-summary",
+    "primary-metrics",
+  ]
+    .concat(Object.keys(primaryMetrics || []))
+    .concat("secondary-metrics")
+    .concat(Object.keys(secondaryMetrics || []))
+    .concat("default-metrics")
+    .concat(Object.keys(analysis?.other_metrics || []));
+
+  return (
+    <Container fluid className="h-100vh" data-testid={testid}>
+      <Row className="h-md-100">
+        <Col
+          md="3"
+          lg="3"
+          xl="2"
+          className="bg-light pt-2 border-right shadow-sm"
+        >
+          <nav
+            data-testid="nav-sidebar"
+            id="scrolling-navbar"
+            className="navbar fixed-top"
+          >
+            <Nav className="flex-column font-weight-semibold" as="ul">
+              <LinkNav
+                storiesOf="pages/Home"
+                className="mb-3 small font-weight-bold"
+                textColor="text-secondary"
+              >
+                <ChevronLeft className="ml-n1" width="18" height="18" />
+                Experiments
+              </LinkNav>
+              <LinkNav
+                route={`${slug}/design`}
+                storiesOf={"pages/Design"}
+                testid={"nav-design"}
+              >
+                <Clipboard className="sidebar-icon" />
+                Design
+              </LinkNav>
+              {analysisAvailable(analysis) ? (
+                <>
+                  <LinkNav
+                    route={`${slug}/results`}
+                    storiesOf={"pages/Results"}
+                    testid={"nav-results"}
+                  >
+                    <BarChart className="sidebar-icon" />
+                    Results
+                  </LinkNav>
+                  <Scrollspy
+                    items={sidebarKeys}
+                    className="text-dark"
+                    currentClassName="font-weight-bold text-primary"
+                  >
+                    <div>
+                      <a href="#monitoring" className="inherit-color">
```

Comment:
I think some of these could use more consistent spacing, but fine to do in a follow up.

<img width="219" alt="image" src="https://user-images.githubusercontent.com/13018240/104668505-d480d800-569d-11eb-82ce-45329de74926.png">

Reply:
I kind of added the extra spacing on purpose between different sections so they're easier to distinguish. Does it seem ugly? I can just remove it :)

Reply:
Meh, we can tweak it later if anyone thinks the same. :P